File: src/api/codex-common.ts
import { invoke } from "./client";
import type { ClientSettingsSnapshot, SettingsValues } from "./settings";
export interface CodexCommonView {
  policy: { version: 1; disabledProfileIds: string[] }; revision: string; settings: ClientSettingsSnapshot;
  fragment: { text: string; revision: string };
}
export const getCodexCommonConfig = (): Promise<CodexCommonView> => invoke("get_codex_common_config");
export const extractCodexCommonConfig = (): Promise<SettingsValues> => invoke("extract_codex_common_config");
export const setCodexCommonConfigEnabled = (profileId: string, enabled: boolean, expectedRevision: string): Promise<CodexCommonView> =>
  invoke("set_codex_common_config_enabled", { profileId, enabled, expectedRevision });
export const saveCodexCommonFragment = (text: string, expectedRevision: string): Promise<CodexCommonView> =>
  invoke("save_codex_common_fragment", { text, expectedRevision });

File: src-tauri/src/commands/codex_common.rs
use super::{
    error::{blocking, state, CommandError},
    ConfigWriteGate,
};
use crate::codex_common::{self, CommonView};
use asb_core::{AppKind, SettingsValues};
use tauri::{AppHandle, Manager};
fn error(message: String) -> CommandError {
    CommandError::new("codex-common-config-invalid", message)
}
#[tauri::command]
pub(crate) async fn get_codex_common_config(app: AppHandle) -> Result<CommonView, CommandError> {
    let state = state(&app)?;
    blocking(move || codex_common::view(&state).map_err(error)).await
}
#[tauri::command]
pub(crate) async fn extract_codex_common_config(
    app: AppHandle,
) -> Result<SettingsValues, CommandError> {
    let state = state(&app)?;
    blocking(move || {
        codex_common::extract(&state.target(AppKind::Codex).map_err(error)?).map_err(error)
    })
    .await
}
#[tauri::command]
pub(crate) async fn set_codex_common_config_enabled(
    app: AppHandle,
    profile_id: String,
    enabled: bool,
    expected_revision: String,
) -> Result<CommonView, CommandError> {
    let state = state(&app)?;
    let gate = app.state::<ConfigWriteGate>().inner().clone();
    blocking(move || {
        let _guard = gate.lock().map_err(error)?;
        codex_common::set_enabled(&state, &profile_id, enabled, &expected_revision).map_err(error)
    })
    .await
}
#[tauri::command]
pub(crate) async fn save_codex_common_fragment(
    app: AppHandle,
    text: String,
    expected_revision: String,
) -> Result<CommonView, CommandError> {
    let state = state(&app)?;
    let gate = app.state::<ConfigWriteGate>().inner().clone();
    blocking(move || {
        let _guard = gate.lock().map_err(error)?;
        codex_common::save_fragment(&state, &text, &expected_revision).map_err(error)?;
        codex_common::view(&state).map_err(error)
    })
    .await
}
